buildkite/resource_cluster_queue.go

```diff
@@ -0,0 +1,210 @@
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"cluster_uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"cluster_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The ID of the Cluster that this Cluster Queue belongs to.",
+			},
+			"key": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The key of the Cluster Queue.",
+			},
+			"description": resource_schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "A description for the Cluster Queue. ",
+			},
+		},
+	}
+}
+
+func (cq *ClusterQueueResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state ClusterQueueResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiResponse, err := createClusterQueue(
+		cq.client.genqlient,
+		cq.client.organizationId,
+		plan.ClusterId.ValueString(),
+		plan.Key.ValueString(),
+		plan.Description.ValueStringPointer(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Cluster Queue",
+			fmt.Sprintf("Unable to create Cluster Queue: %s", err.Error()),
+		)
+		return
+	}
+
+	state.Id = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Id)
+	state.Uuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Uuid)
+	state.ClusterId = plan.ClusterId
+	state.ClusterUuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Cluster.Uuid)
+	state.Key = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Key)
+	state.Description = types.StringPointerValue(&apiResponse.ClusterQueueCreate.ClusterQueue.Description)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (cq *ClusterQueueResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ClusterQueueResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	queues, err := getClusterQueues(cq.client.genqlient, cq.client.organization, state.ClusterUuid.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read Cluster Queue",
+			fmt.Sprintf("Unable to read Cluster Queue: %s", err.Error()),
+		)
+		return
+	}
+
+	// Find the Cluster Q from the returned queues to update state
+	for i := range queues.Organization.Cluster.Queues.Edges {
+		if queues.Organization.Cluster.Queues.Edges[i].Node.Id == state.Id.ValueString() {
+			// Update ClusterQueueResourceModel with Node values and append
+			updateClusterQueueResource(queues.Organization.Cluster.Queues.Edges[i].Node, &state)
+			resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+			break
+		}
+	}
```

Comment:
You can make this a bit nicer to read

```suggestion
	for i, edge := range queues.Organization.Cluster.Queues.Edges {
		if edge.Node.Id == state.Id.ValueString() {
			// Update ClusterQueueResourceModel with Node values and append
			updateClusterQueueResource(edge.Node, &state)
			resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
			break
		}
	}
```

Reply:
I doubt we will hit it but there is the chance that 50 cluster queues is not enough. I don't think we need to code for it right now, but at least logging if we go through all 50 queues and don't find the correct one, error out. And if we get reports of hitting that error, we can fix it then

Reply:
Added a check to the bottom of `Read` to log that case - more a tradeoff until the fetching of 50 queues in a cluster can be switched to a `node` or a `clusterQueue(id: "x")` query